### PR TITLE
[CELEBORN-277] PushDataHandle callback could miss soft split status

### PR DESCRIPTION
**Pull request by boneanxs**
### What changes were proposed in this pull request?

Return soft split status if soft split is true

### Why are the changes needed?

PushDataHandle callback could miss soft split status

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?
Existing UT.

### CODE REVIEW

File: worker/src/main/scala/org/apache/celeborn/service/deploy/worker/PushDataHandler.scala [hunk 1]
```diff
@@ -357,22 +357,27 @@ class PushDataHandler extends BaseMessageHandler with Logging {
...
       // 2. the client enables push data to the replica, and the replica worker could hit here
-      Option(CongestionController.instance()) match {
-        case Some(congestionController) =>
-          if (congestionController.isUserCongested(fileWriter.getFileInfo.getUserIdentifier)) {
-            if (isMaster) {
-              callbackWithTimer.onSuccess(
-                ByteBuffer.wrap(
-                  Array[Byte](StatusCode.PUSH_DATA_SUCCESS_MASTER_CONGESTED.getValue)))
+      if (softSplit.get()) {
+        callbackWithTimer.onSuccess(ByteBuffer.wrap(Array[Byte](StatusCode.SOFT_SPLIT.getValue)))
```
Comment (AngersZhuuuu) [inline worker/src/main/scala/org/apache/celeborn/service/deploy/worker/PushDataHandler.scala:361]: Better to add some comment's to show if we return soft split , won't return congestion ect..

**Review by AngersZhuuuu — APPROVED** [review]
LGTM

**Comment by AngersZhuuuu** [comment]
ping @waitinfuture

File: worker/src/main/scala/org/apache/celeborn/service/deploy/worker/PushDataHandler.scala [hunk 2]
```diff
@@ -285,6 +285,10 @@ class PushDataHandler extends BaseMessageHandler with Logging {
                 resp.flip()
                 callbackWithTimer.onSuccess(resp)
               } else if (softSplit.get()) {
+                // TODO Currently if the worker is in soft split status, given the guess that the client
+                // will fast stop pushing data to the worker, we won't return congest status. But
+                // in the long term, especially if this issue could frequently happen, we may need to return
+                // congest&softSplit status together
```
Comment (waitinfuture) [inline worker/src/main/scala/org/apache/celeborn/service/deploy/worker/PushDataHandler.scala:291]: Agree with the comment, in the future we might encode multiple status in the response.

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks!

**Pull request merged**